cmake/VASTRegisterPlugin.cmake

```diff
@@ -272,11 +272,7 @@ macro (VASTInstallExampleConfiguration target source prefix destination)
 
   # Set a temporary variable for the example dir location. Because we're in a
   # macro we're unsetting the variable again later on.
-  if (VAST_CMAKE_INSTALL_DOCDIR)
-    set(_example_dir "${VAST_CMAKE_INSTALL_DOCDIR}/examples")
-  else ()
-    set(_example_dir "${CMAKE_INSTALL_DOCDIR}/examples")
-  endif ()
```

Comment:
Why do we need this change? Can we also remove VAST_CMAKE_INSTALL_DOCDIR from VASTConfig.cmake.in?

Reply:
It is explained in the commit message:

> Before this change example configs and READMEs of out of tree
plugins would always installed into the docs location of the main
installation. If this is desired we can still achieve it by setting
`CMAKE_INSTALL_DOCDIR` explicitly when builiding the plugin.
>
> This change exists for two reasons:
> 1. It doesn't really make sense to install a plugin into one place
>    but the docs for it somewhere else.
> 2. When building out of tree plugins with Nix the main installation
>    tree of VAST is read-only, so the build would fail.